tests/src/test/scala/org/apache/openwhisk/core/limits/ActionLimitsTests.scala

```diff
@@ -463,4 +465,38 @@ class ActionLimitsTests extends TestHelpers with WskTestHelpers with WskActorSys
       _.response.result.get.fields("error") shouldBe Messages.memoryExhausted.toJson
     }
   }
+
+  /**
+   * Test that a heavy logging action is interrupted within its timeout limits.
+   */
+  it should "interrupt the heavy logging action within its time limits" in withAssetCleaner(wskprops) {
+    (wp, assetHelper) =>
+      val name = s"NodeJsTestLoggingActionCausingTimeout-${System.currentTimeMillis()}"
+      print(s"\n create action ${name} using api host: ${wskprops.apihost}..")
+      assetHelper.withCleaner(wsk.action, name, confirmDelete = true) { (action, _) =>
+        action.create(
+          name,
+          Some(TestUtils.getTestActionFilename("loggingTimeout.js")),
+          timeout = Some(allowedActionDuration))
+      }
+      val duration = allowedActionDuration + 3.minutes
+      val checkDuration = allowedActionDuration + 1.minutes
+      val run =
+        wsk.action.invoke(name, Map("durationMillis" -> duration.toMillis.toJson, "delayMillis" -> 100.toJson))
+      withActivation(wsk.activation, run) { result =>
+        withClue("Activation result not as expected:") {
+          result.response.status shouldBe ActivationResponse.messageForCode(ActivationResponse.DeveloperError)
+          result.response.result.get.fields("error") shouldBe {
+            Messages.timedoutActivation(allowedActionDuration, init = false).toJson
+          }
+          val logs = result.logs.get
+          logs.last should include(Messages.logFailure)
+
+          val parseLogTime = (line: String) => Instant.parse(line.split(' ').head)
+          val startTime = parseLogTime(logs.head)
+          val endTime = parseLogTime(logs.last)
+          between(startTime, endTime).asInstanceOf[FiniteDuration] should be < checkDuration
```

Comment:
Is this `.asInstanceOf[FiniteDuration]` really needed?

Reply:
We tried without, but got `Cannot resolve reference with such as signature`, so we tried with `asInstanceOf[FiniteDuration]` which worked in intellij, but during compile with grade we get now a compile error, so we still trying to solve this problem.
```
/home/vagrant/github.ibm.com/bluewhisk/open/tests/src/test/scala/org/apache/openwhisk/core/limits/ActionLimitsTests.scala:498: diverging implicit expansion for type scala.concurrent.duration.FiniteDuration => Comparable[scala.concurrent.duration.FiniteDuration]
starting with method $conforms in object Predef
          between(startTime, endTime).asInstanceOf[FiniteDuration] should be < checkDuration
```

Reply:
We finally changed the implementation back to millis because we cannot compare `FiniteDuration` with `should be <`. If someone has an idea how to solve this problem please let us know.

```
> Task :tests:compileTestScala
...
/home/vagrant/github.ibm.com/bluewhisk/open/tests/src/test/scala/org/apache/openwhisk/core/limits/ActionLimitsTests.scala:496: diverging implicit expansion for type scala.concurrent.duration.FiniteDuration => Comparable[scala.concurrent.duration.FiniteDuration]
starting with method $conforms in object Predef
          between(startTime, endTime) should be < checkDuration
                                                ^
```

Reply:
@markusthoemmes I think we should accept the latest results and proposal by @steffenrost @ruediger-maass and compare the `Duration`s millis instead of the `Duration`s millis.